src/cli/proxy.rs

```diff
@@ -115,13 +115,22 @@ impl Proxy {
             None
         };
 
-        self.run_recv_from(&config, sessions, shutdown_rx.clone())?;
+        self.run_recv_from(&config, sessions.clone(), shutdown_rx.clone())?;
         tracing::info!("Quilkin is ready");
 
         shutdown_rx
             .changed()
             .await
-            .map_err(|error| eyre::eyre!(error))
+            .map_err(|error| eyre::eyre!(error))?;
+
+        tracing::info!(sessions=%sessions.len(), "waiting for active sessions to expire");
+        while sessions.is_not_empty() {
+            tokio::time::sleep(SESSION_TIMEOUT_SECONDS).await;
```

Comment:
Non blocking suggestion:

```suggestion
            tokio::time::sleep(Duration::from_secs(1)).await;
```
?

I just figured if we're checking every 60 seconds we could potentially shoot over by >=59 seconds from when we actually have all sessions expired. I figure no harm in checking one a secord to see if it's finally empty - and then you'll end up shutting down pretty quick after the session map is empty.